thunder/torch/__init__.py

```diff
@@ -170,23 +170,18 @@ def is_floating_point(a: TensorLike, /) -> bool:
 
 
 # Handles the size method
-def size(a):
-    def fn_(idx: int | None = None):
-        if idx is None:
-            return a.shape
-        return a.shape[idx]
-
-    return fn_
+@torchsymbol(torch.Tensor.size, is_method=True, id="torch.Tensor.size")
+def size(a: TensorLike, dim: None | int = None) -> int | TensorLike:
```

Comment:
Nit: 

```
(a: TensorLike, /, dim: None | int = None) -> int | Sequence[int]...
```

The shape isn't stored as a tensor, but a sequence of integers (it actually might always be a tuple of integers)

fyi @jjsjann123

Reply:
Fixed